arch/arc/core/fast_irq.S

```diff
@@ -22,6 +22,23 @@
 GTEXT(_firq_enter)
 GTEXT(_firq_exit)
 
+#if defined(CONFIG_ARC_FIRQ_STACK)
+GTEXT(z_arc_firq_stack_set)
+SECTION_FUNC(TEXT, z_arc_firq_stack_set)
+	clri r1
```

Comment:
the z_arc_firq_stack_set must be called when irq diasbled,  it may be called not only in the init phase but also other places.  So clri r1 and seti r1 are needed.  seti r1 may not enable irq but just restore the original IE status.

Reply:
Then please explain this in comments so that changes I propose are  not implemented by somebody else later :)

Reply:
add comments for this

Reply:
comments added